service/src/main/java/uk/gov/hmcts/reform/fpl/service/cafcass/api/CafcassApiCaseDocumentsConverter.java

```diff
@@ -0,0 +1,257 @@
+import uk.gov.hmcts.reform.fpl.model.PlacementSupportingDocument;
+import uk.gov.hmcts.reform.fpl.model.Supplement;
+import uk.gov.hmcts.reform.fpl.model.SupportingEvidenceBundle;
+import uk.gov.hmcts.reform.fpl.model.cafcass.api.CafcassApiCaseData;
+import uk.gov.hmcts.reform.fpl.model.cafcass.api.CafcassApiCaseDocument;
+import uk.gov.hmcts.reform.fpl.model.common.C2DocumentBundle;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.OtherApplicationsBundle;
+import uk.gov.hmcts.reform.fpl.model.common.SubmittedC1WithSupplementBundle;
+import uk.gov.hmcts.reform.fpl.model.group.C110A;
+import uk.gov.hmcts.reform.fpl.model.interfaces.WithDocument;
+import uk.gov.hmcts.reform.fpl.model.order.DraftOrder;
+import uk.gov.hmcts.reform.fpl.model.order.HearingOrder;
+import uk.gov.hmcts.reform.fpl.model.order.generated.GeneratedOrder;
+import uk.gov.hmcts.reform.fpl.service.document.ManageDocumentService;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.AA_PARENT_ORDERS;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.C1_APPLICATION_DOCUMENTS;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.C2_APPLICATION_DOCUMENTS;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.NOTICE_OF_ACTING_OR_ISSUE;
+import static uk.gov.hmcts.reform.fpl.enums.cfv.DocumentType.PLACEMENT_RESPONSES;
+import static uk.gov.hmcts.reform.fpl.utils.DocumentsHelper.getDocumentIdFromUrl;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CafcassApiCaseDocumentsConverter implements CafcassApiCaseDataConverter {
+    private final ManageDocumentService manageDocumentService;
+
+    @Override
+    public CafcassApiCaseData.CafcassApiCaseDataBuilder convert(CaseData caseData,
+                                                                CafcassApiCaseData.CafcassApiCaseDataBuilder builder) {
+        return builder.caseDocuments(getCaseDocuments(caseData));
+    }
+
+    private List<CafcassApiCaseDocument> getCaseDocuments(CaseData caseData) {
+        return Stream.of(
+                getStandardAndUrgentDirectionOrder(caseData),
+                getDraftOrders(caseData),
+                getApprovedOrders(caseData),
+                getOriginalApplications(caseData),
+                getPlacementApplications(caseData),
+                getAdditionalApplications(caseData),
+                getHearingNotice(caseData),
+                getManageDocuments(caseData))
+            .flatMap(List::stream)
+            //.parallel()
```

Comment:
should we keep this in?

Reply:
we don't have to do it parallel if the performance is good enough already.